platform/o.n.bootstrap/src/org/netbeans/Util.java

```diff
@@ -170,6 +171,9 @@ public static boolean checkPackageDependency(Dependency dep, ClassLoader cl) thr
             } else {
                 pkg = Package.getPackage(packageName);
             }
+            if (pkg == null && sampleClass != null) {
+                pkg = sampleClass.getPackage();
```

Comment:
I think this is correct behavior. We load the `sampleClass` and we are interested in its package. Not sure why used to be done in such a complicated way by throwing the `sampleClass` away. Anyway, @jlahoda wanted to know why it is working in the IDE and it is broken in the test. The `getPackage()` method traverses the `parent` classloaders and here is their list from bottom most to the application classloader:
```
jdk.internal.loader.ClassLoaders$PlatformClassLoader
org.netbeans.junit.NbModuleSuite$JUnitLoader
java.net.URLClassLoader
org.netbeans.Module$StandardClassLoader[debugger.jpda]
```
the problem is probably caused by the `JUnitLoader` as it does its best to skip the application classloader (to force module classes to be loaded by module classloaders) and as 
```
l.loadClass("com.sun.jdi.VirtualMachineManager").getClassLoader()
$5 ==> jdk.internal.loader.ClassLoaders$AppClassLoader@5bc2b487
```
it is not found.

Reply:
Thanks for the investigation!